core/domain/auth_domain.py

```diff
@@ -31,3 +33,34 @@
 # corresponding Oppia-generated IDs in our APIs.
 AuthIdUserIdPair = (
     collections.namedtuple('AuthIdUserIdPair', ['auth_id', 'user_id']))
+
+
+class AuthClaims(python_utils.OBJECT):
+    """Domain object representing essential Claims (a piece of information about
+    a user, like name, address, phone number, etc.) of an authorized user.
+
+    Attributes:
+        auth_id: str|None. A unique identifier associated to the user. The ID is
```

Comment:
Why can this be None? In general, when None is an acceptable value, its significance should be explained.

Ditto for email.

Reply:
It can't, so I decided to raise an error instead. Done (+tests).